library-core/src/main/java/com/mikepenz/fastadapter/MutableSubItemList.kt

```diff
@@ -43,4 +43,14 @@ class MutableSubItemList<E : ISubItem<*>>(val parent: IParentItem<*>, val list:
         element.parent = parent
         return list.set(index, element).also { oldElement -> oldElement.parent = null }
     }
+
+    override fun clear() {
+        list.forEach { it.parent = null }
+        list.clear()
+    }
+
+    fun setNewList(newList: List<E>) {
```

Comment:
set new list should not clear and add, but instead actually cause a complete new list reference

Reply:
changed to re-link to reference of new list